plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/CloseableInMemoryRecordCursor.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.spi.connector.InMemoryRecordSet;
+import io.trino.spi.type.Type;
+import org.apache.iceberg.io.CloseableIterator;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+
+public class CloseableInMemoryRecordCursor
+        extends InMemoryRecordSet.InMemoryRecordCursor
+{
+    private boolean closed;
+
+    public CloseableInMemoryRecordCursor(List<Type> types, Iterable<? extends List<?>> records)
```

Comment:
Why not provide a `CloseableIterator<? extends List<?>>` instead of providing the whole `Iterable<? extends List<?>>` ?